### DISCUSSION TITLE
Show error output of Mermaid diagrams within note?

### DISCUSSION THREAD

**Discussion by perfectra1n** [TriliumNext/Trilium#3610] [Q&A] [answered]
Hi Trilium community, zadam,

Just wanted to say that I love the app. I've used it for years now and shill it to my friends/family lol.

I recently got into Mermaid diagrams more, and I was surprised to see that on sites like Mermaid.live it shows the error of the diagram:
![image](https://user-images.githubusercontent.com/2390633/218571336-54ca6bfd-447b-4333-8565-f7f6de27bbe0.png)

However, when doing the same in Trilium, it doesn't show the error and just displays the generic "Syntax error":
![image](https://user-images.githubusercontent.com/2390633/218571413-2476e40c-dd89-4bd7-b752-dc394ec18b7a.png)

Is there a way to show the syntax error in the GUI / note pane? I believe that mermaid.js is being utilized to show the Mermaid diagram, and I believe that they have a `parseError` function [here](https://mermaid.js.org/config/usage.html#advanced-usage).

Thanks for looking at my question, again, I love Trilium and I've used it to finish college, and climb the career ladder, so it's amazing!

**Comment by zadam** [comment, marked as answer]
Fixed, actually this was supported earlier, but broke probably in some Mermaid upgrade:

![image](https://user-images.githubusercontent.com/617641/218591722-ff49e5c3-e108-4829-9b6f-10c1e93de7d7.png)

**Reply by perfectra1n** [reply]
Oh dang, thank you so much! I appreciate you fixing that so quickly! You're the best :)
